src/transformers/tokenization_utils.py

```diff
@@ -1950,51 +1953,67 @@ def batch_encode_plus(
 
     def encode_plus(
         self,
-        text,
-        text_pair=None,
-        add_special_tokens=True,
-        max_length=None,
-        pad_to_max_length=False,
-        stride=0,
-        truncation_strategy="longest_first",
-        return_tensors=None,
-        return_token_type_ids=True,
-        return_attention_mask=True,
-        return_overflowing_tokens=False,
-        return_special_tokens_mask=False,
-        return_offsets_mapping=False,
+        text: Union[str, List[str]],
+        text_pair: Optional[str] = None,
+        add_special_tokens: bool = True,
+        max_length: Optional[int] = None,
+        pad_to_max_length: bool = False,
+        stride: int = 0,
+        truncation_strategy: str = "longest_first",
+        return_tensors: Optional[str] = None,
+        return_token_type_ids: bool = True,
+        return_attention_mask: bool = True,
```

Comment:
You should rebase on master if you can because I think this is now Optional[bool] = None since @LysandreJik worked to adapt the output to the models.

Reply:
I'm not sure to fully undestand why bool should be Optional ? A default value would be more understable imho and remove the need for None checking, wdyt ?

Reply:
You should check w @LysandreJik :)

Reply:
The spirit behind having those values as `None` by default is the following: if this value is `None`, then all those returns are set to the default *tokenizer-specific values*. This is different from tokenizer to tokenizer, e.g. DistilBERT should by default return `input_ids` and `attention_mask`, but **not** `token_type_ids`, as the model cannot handle it. This in turn allows the user to do the following:

```py
inputs = tokenizer.encode_plus(values, return_tensors="pt")
model(**inputs)
```

And this now works with every model.

These values can still be explicitly set to `True` or `False` by the user. See #3116 for more information/implementation details.